cloudimg/ms_azure.py

```diff
@@ -285,11 +287,15 @@ def filter_object_by_tags(self, tags):
                 blobs_list = container_client.find_blobs_by_tags(
                     filter_expression=filter_expression
                 )
-                try:
-                    return blobs_list.next()
-                except StopIteration:
-                    continue
-        return
+                for tag in tags.keys():
```

Comment:
The concept of this loop doesn't make much sense to me.  It can be summarized as "increment an iterator once for each tag".  This only seems to make sense if, when called with N tags, you know that find_blobs_by_tag should return N blobs, and furthermore the log messages only make sense if those were returned in the same order as the input tags.  Is that really how it works?

More detailed comments inline.